src/Button/Button.component.tsx

```diff
@@ -567,10 +552,33 @@ export const Button = ({
 
     return (
         <StyledButton
+            onMouseDown={event => {
+                setMouseDown(true);
+                if (onMouseDown) {
+                    onMouseDown(event);
+                }
+            }}
+            onMouseUp={event => {
+                setMouseDown(false);
+                if (onMouseUp) {
+                    onMouseUp(event);
+                }
+            }}
+            onFocus={event => {
+                if (mouseDown) {
+                    // This keeps the button from being :focused when
+                    // clicked so that it is only applied when tabbed to.
+                    // We want the outline to appear when tabbing for
+                    // accessibility.
+                    event.target.blur();
```

Comment:
Good note. The above methods could be consolidated into a single handler with a binary flag.

Reply:
Also, TS may flip out over the absence of type declaration on the `event` arg. Surprised your linter didn't scream bloody murder.